arcade/examples/sprite_health.py

```diff
@@ -1,230 +1,297 @@
 """
-import random
+import math
+from typing import Tuple
+
 import arcade
+from arcade.resources import (
+    image_female_person_idle,
+    image_laser_blue01,
+    image_zombie_idle,
+)
 
 SPRITE_SCALING_PLAYER = 0.5
-SPRITE_SCALING_COIN = 0.2
-SPRITE_SCALING_LASER = 0.8
-COIN_COUNT = 50
+SPRITE_SCALING_ENEMY = 0.5
+SPRITE_SCALING_BULLET = 1
+INDICATOR_BAR_OFFSET = 32
+ENEMY_ATTACK_COOLDOWN = 1
+BULLET_SPEED = 150
+BULLET_DAMAGE = 1
+PLAYER_HEALTH = 5
 
 SCREEN_WIDTH = 800
 SCREEN_HEIGHT = 600
-SCREEN_TITLE = "Hit Points and Health Bars "
-
-BULLET_SPEED = 5
-
-HEALTHBAR_WIDTH = 25
-HEALTHBAR_HEIGHT = 3
-HEALTHBAR_OFFSET_Y = -10
-
-HEALTH_NUMBER_OFFSET_X = -10
-HEALTH_NUMBER_OFFSET_Y = -25
-
-
-class SpriteWithHealth(arcade.Sprite):
-    """ Sprite with hit points """
-
-    def __init__(self, image, scale, max_health):
-        super().__init__(image, scale)
-
-        # Add extra attributes for health
-        self.max_health = max_health
-        self.cur_health = max_health
-
-    def draw_health_number(self):
-        """ Draw how many hit points we have """
-
-        health_string = f"{self.cur_health}/{self.max_health}"
-        arcade.draw_text(health_string,
-                         start_x=self.center_x + HEALTH_NUMBER_OFFSET_X,
-                         start_y=self.center_y + HEALTH_NUMBER_OFFSET_Y,
-                         font_size=12,
-                         color=arcade.color.WHITE)
-
-    def draw_health_bar(self):
-        """ Draw the health bar """
-
-        # Draw the 'unhealthy' background
-        if self.cur_health < self.max_health:
-            arcade.draw_rectangle_filled(center_x=self.center_x,
-                                         center_y=self.center_y + HEALTHBAR_OFFSET_Y,
-                                         width=HEALTHBAR_WIDTH,
-                                         height=3,
-                                         color=arcade.color.RED)
-
-        # Calculate width based on health
-        health_width = HEALTHBAR_WIDTH * (self.cur_health / self.max_health)
-
-        arcade.draw_rectangle_filled(center_x=self.center_x - 0.5 * (HEALTHBAR_WIDTH - health_width),
-                                     center_y=self.center_y - 10,
-                                     width=health_width,
-                                     height=HEALTHBAR_HEIGHT,
-                                     color=arcade.color.GREEN)
+SCREEN_TITLE = "Sprite Health Bars"
+
+
+def sprite_off_screen(
+    sprite: arcade.Sprite,
+    screen_height: int = SCREEN_HEIGHT,
+    screen_width: int = SCREEN_WIDTH,
+) -> bool:
+    """Checks if a sprite is off-screen or not."""
+    return (
+        sprite.top < 0
+        or sprite.bottom > screen_height
+        or sprite.right < 0
+        or sprite.left > screen_width
+    )
+
+
+class Player(arcade.Sprite):
+    def __init__(self, bar_list: arcade.SpriteList) -> None:
+        super().__init__(filename=image_female_person_idle, scale=SPRITE_SCALING_PLAYER)
```

Comment:
When examples are built as html doc, this line currently extends beyond the right boundary of the embedded view. Splitting the line up would help.
```suggestion
        super().__init__(
            filename=image_female_person_idle,
            scale=SPRITE_SCALING_PLAYER
        )
```